sweetalert2.d.ts

```diff
@@ -356,10 +363,12 @@ declare module 'sweetalert2' {
 
         /**
          * If set to false, the user can't dismiss the modal by clicking outside it.
+         * You can also pass a custom function returning a boolean value, e.g. if you want
+         * to disable outside clicks for the loading state of a modal.
          *
          * @default true
          */
-        allowOutsideClick?: boolean;
+        allowOutsideClick?: boolean | SweetAlertAllowOutsideClick;
```

Comment:
@limonte It's hard to tell intuitively what a `SweetAlertAllowOutsideClick` is.. Is there a more descriptive name that could be used? Do we *need* this named? Or could we just use the type `boolean | () => boolean`?

I'm working on #793 now. I'll save the typings for last.

Reply:
Or maybe the typings file is best left to @toverux

Reply:
Yes it is preferable to not create a named type here, since this one won't be reused and only serves for allowOutsideClick.